### DISCUSSION TITLE
Are there any downsides to the keyword extraction feature?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1350] [Q&A]
Without specifying `word`, running `tree-sitter generate` on [my grammar](https://github.com/tlaplus-community/tree-sitter-tlaplus/) takes 16 seconds and generates a 30 MB parser.c file. Afterward, it also takes 16 seconds but only generates a 27.5 MB parser.c file. Sure it makes the parser debug output look a little weird since it reads every keyword twice, like the following for the string `IF`:
> lex_internal state:183, row:1, column:5
  skip character:' '
  consume character:'I'
  consume character:'F'
  consume character:'I'
  consume character:'F'

and maybe that comes with performance issues. It also makes the parser less permissive, which maybe you don't want if you're using it for language tooling. But other than that are there any downsides?

**Comment by ObserverOfTime** [comment]
Probably not. :)
